lang/pypy/Portfile

```diff
@@ -244,26 +264,27 @@ post-patch {
 # intensive, and one of the best cases for using PyPy itself over
 # CPython.
 if {$subport ne $name} {
-    # just use our own pypy2 for building pypy3
+    # just use our own pypy3.x for building other pypy3.x subports
     depends_build-append port:pypy
-    build.cmd       ${prefix}/lib/pypy/bin/pypy
+    build.cmd       ${prefix}/lib/pypy/bin/pypy-${python.branch}
 } else {
     # use a binary distribution of pypy itself
     checksums-prepend       ${distname}${extract.suffix}
 
     if {${os.major} >= 20} {
         # older versions cannot import C extensions on macOS 11 and later
-        set bootstrapper        "pypy2.7-v${version}-osx64"
-        distfiles-append        ${bootstrapper}${extract.suffix}
+        set bootstrapper        "pypy2.7-v7.3.20-macos_arm64"
+        distfiles-append        ${bootstrapper}.tar.bz2:bootstrap
```

Comment:
This port is configured to use .tar.gz distfiles; MacPorts doesn't support mixing and matching different distfile compression methods with its default extract method. It happens to work on OS X 10.9 and later because /usr/bin/gzip and /usr/bin/bzip2 both support gz and bz2 and xz, but it will fail on OS X 10.8 and earlier. Please implement the usual workaround: set `extract.only` to the list of distfiles that should use the default extraction method (for .tar.gz) and then use a `post-extract` block to manually extract the other file(s).